src/core/solvers/ddp.cpp

```diff
@@ -402,11 +403,19 @@ void SolverDDP::allocateData() {
   fTVxx_p_ = Eigen::VectorXd::Zero(ndx);
 }
 
-const double& SolverDDP::get_regfactor() const { return regfactor_; }
+const double& SolverDDP::get_reg_incfactor() const { return reg_incfactor_; }
```

Comment:
There is no reason to return a cont ref here for a double.

Reply:
This applies everywhere!

Reply:
Could explain me the reason of this briefly?
What about using `const double&` or `const bool&` as in/out arguments of a function?

Reply:
You have a larger overhead when using const ref for small objects such as bool or double. There is no reason to not return them by copy, as the memory will lower or equal.

Reply:
Done for all the solvers and shooting problem!